coral/src/app/features/topics/request/schemas/topic-request-form.ts

```diff
@@ -0,0 +1,198 @@
+import z, { RefinementCtx } from "zod";
+import isNumber from "lodash/isNumber";
+import { UseFormReturn } from "react-hook-form";
+import { useEffect } from "react";
+
+const topicNameField = z
+  .string()
+  .min(1, { message: "Topic name can not be empty" })
+  .max(255, { message: "Topic name length can not exceed 255 characters" });
+
+const topicpartitionsField = z.string();
+
+const replicationfactorField = z.string();
+
+const environmentField = z.object({
```

Comment:
Do we hand type every schema, or do we have a tool for generating them from the TS types? If we don't have one, should we get one? :o

Reply:
At the moment yes. We could look into some generator. However, in this specific case, I would not like to have the whole `Environment` as the type because we do not need all properties.

Reply:
I would expect we would generate those `zod` schemas from the types exported by `domain` (even if they also might not fit), or at the moment we need them.

But also, it's kind of backwards of what we are currently doing, writing the `zod` schema and then using `.infer` to generate the types.

We'll see if it becomes an actual issue or if I'm just imagining it haha